### PR TITLE
Add option to specify hostname in BrowserRefreshServer

### PR DESCRIPTION
**Pull request by zbecknell**
Summary of the changes
 - Looks for an environment variable specifying the hostname for the dotnet-watch browser refresh server.

Addresses #25570

### CODE REVIEW

**Comment by javiercn** [comment]
@zbecknell thanks for the contribution.

The change looks good I think, but I'll let @pranavkm judge it.

@Pilchie is this something you would be ok with us taking it in RC2?

**Comment by Pilchie** [comment]
I'll consider it through the end of next week, but I don't think it's worth taking to Tactics for approval.

**Comment by Pilchie** [comment]
(waiting for @pranavkm to sign off though).

**Comment by javiercn** [comment]
@zbecknell did you try to re-target the branch or did you add commits on top of your previous commit?

**Comment by javiercn** [comment]
You should be targeting release/5.0-rc2 for now, in case @pranavkm comes back an is ok with the change, `master` will be 6.0 I believe.

**Comment by zbecknell** [comment]
Ahh, man, I screwed this branch up, let me revert and target the proper branch as well.  Shouldn't have been working in master!

**Comment by javiercn** [comment]
@zbecknell don't worry 😄 we've all ben there. git cherry-pick is your friend 😃

**Comment by zbecknell** [comment]
Ug... sorry about all the reviewers added... I retargeted without resetting my `master` to rc2, causing a bunch of commits to show up for a bit.

File: src/Tools/dotnet-watch/src/BrowserRefreshServer.cs [hunk 1]
```diff
@@ -33,11 +33,13 @@ public BrowserRefreshServer(IReporter reporter)
 
         public async ValueTask<string> StartAsync(CancellationToken cancellationToken)
         {
+            var hostName = Environment.GetEnvironmentVariable("ASPNETCORE_AUTO_RELOAD_WS_HOSTNAME") ?? "127.0.0.1";
```
Comment (pranavkm) [inline src/Tools/dotnet-watch/src/BrowserRefreshServer.cs:36]: @zbecknell the dotnet-watch arguments tend to use the DOTNET_WATCH prefix. Could you update it to use that instead?

```suggestion
            var hostName = Environment.GetEnvironmentVariable("DOTNET_WATCH_AUTO_RELOAD_WS_HOSTNAME") ?? "127.0.0.1";
```

**Review by mkArtakMSFT — APPROVED** [review]
Thanks for your contribution, @zbecknell !

**Comment by pranavkm** [comment]
@Pilchie could we approve this for rc2?

**Review by pranavkm — APPROVED** [review]

**Comment by Pilchie** [comment]
Approved for .NET 5 RC2 - thanks for the contribution @zbecknell

**Pull request merged**